src/System.Net.Primitives/src/System/Net/IPAddressParser.cs

```diff
@@ -21,22 +34,34 @@ internal static unsafe IPAddress Parse(string ipString, bool tryParse)
                 }
                 throw new ArgumentNullException(nameof(ipString));
             }
+ 
+            return Parse(ipString.ToCharArray(), tryParse);
```

Comment:
Ouch, we should not be allocating a char[] from the string.  Once you've added the System.Memory reference, you can use the AsReadOnlySpan extension method.